src/core/data/ndarray/ndarray_vector.h

```diff
@@ -134,12 +146,7 @@ namespace core
     template<std::size_t dim>
     auto makeNdArray(std::array<std::uint32_t, dim> sizes)
     {
-        if constexpr (dim == 1)
-            return NdArrayVector<1>{sizes[0]};
-        if constexpr (dim == 2)
-            return NdArrayVector<2>{sizes[0], sizes[1]};
-        if constexpr (dim == 3)
-            return NdArrayVector<3>{sizes[0], sizes[1], sizes[2]};
+        return NdArrayVector<dim>{sizes};
```

Comment:
this function (makeNdArray) seems kinda redundant now

Reply:
did you try to delete it?

Reply:
nope but I will I think

Reply:
removed